core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -613,6 +634,12 @@ final class ShuffleBlockFetcherIterator(
           }
           if (isNetworkReqDone) {
             reqsInFlight -= 1
+            if (!buf.isInstanceOf[NettyManagedBuffer]) {
```

Comment:
> Non-`NettyManagedBuffer` doesn't occupy Netty's memory 
For me this means we do not necessary have some extra free off-heap memory by this.

Could you please extend this comment with reasoning behind why the `isNettyOOMOnShuffle` flag is set to `false` in this case?

Reply:
> For me this means we do not necessary have some extra free off-heap memory by this.

Yes. that's true.

(So the original assuming here is that whenever there's a completed request, there would be some freed memory.)

I have another idea here: https://github.com/apache/spark/pull/32287#discussion_r620779976

Could you also take a look?